.dotnet/src/Custom/Chat/ChatTool.cs

```diff
@@ -68,21 +68,24 @@ internal ChatTool(InternalFunctionDefinition function)
     /// </remarks>
     public BinaryData FunctionParameters => Function?.Parameters;
 
+    public bool? UseStrictSchema => Function?.Strict;
+
     // CUSTOM: Added custom constructor.
     /// <summary>
     /// Creates a new instance of <see cref="ChatTool"/>.
     /// </summary>
     /// <param name="functionName"> The <c>name</c> of the function. </param>
     /// <param name="functionDescription"> The <c>description</c> of the function. </param>
     /// <param name="functionParameters"> The <c>parameters</c> into the function, in JSON Schema format. </param>
-    public static ChatTool CreateFunctionTool(string functionName, string functionDescription = null, BinaryData functionParameters = null)
+    public static ChatTool CreateFunctionTool(string functionName, string functionDescription = null, BinaryData functionParameters = null, bool? useStrictSchema = null)
```

Comment:
Does something funny happen when we have no `parameters` (i.e., no JSON schema) but set `strict` to `false`?

Reply:
I just added a fairly comprehensive test to cover the combinations -- outside of quirkiness when using an older model (`gpt-4o` emits empty structure; you have to explicitly request `gpt-4o-2024-08-06`) it behaves as you'd intuitively hope. The only combination of parameters that results in an error is the one that makes sense, which is providing no schema but requesting a strict match.